zarr/v3/group.py

```diff
@@ -107,73 +116,276 @@ def from_json(
         )
         return group
 
-    @classmethod
-    async def open_or_array(
-        cls,
-        store: StoreLike,
-        runtime_configuration: RuntimeConfiguration = RuntimeConfiguration(),
+    async def getitem(
+        self,
+        key: str,
     ) -> Union[Array, Group]:
-        store_path = make_store_path(store)
-        zarr_json_bytes = await (store_path / ZARR_JSON).get_async()
-        if zarr_json_bytes is None:
-            raise KeyError
-        zarr_json = json.loads(zarr_json_bytes)
-        if zarr_json["node_type"] == "group":
-            return cls.from_json(store_path, zarr_json, runtime_configuration)
-        if zarr_json["node_type"] == "array":
-            return Array.from_json(
-                store_path, zarr_json, runtime_configuration=runtime_configuration
+
+        store_path = self.store_path / key
+
+        if self.zarr_format == 3:
+            zarr_json_bytes = await (store_path / ZARR_JSON).get_async()
+            if zarr_json_bytes is None:
+                # implicit group?
+                zarr_json = {
+                    "zarr_format": self.zarr_format,
+                    "node_type": "group",
+                    "attributes": {},
+                }
+            else:
+                zarr_json = json.loads(zarr_json_bytes)
+            if zarr_json["node_type"] == "group":
+                return type(self).from_json(store_path, zarr_json, self.runtime_configuration)
+            if zarr_json["node_type"] == "array":
+                return Array.from_json(
+                    store_path, zarr_json, runtime_configuration=self.runtime_configuration
+                )
+        elif self.zarr_format == 2:
+            # Q: how do we like optimistically fetching .zgroup, .zarray, and .zattrs?
```

Comment:
I think the overfetching is fine because it is happening in parallel.

Reply:
I was thinking about this issue the other day. One possible behavior when you "open" anything in Zarr would be to call the `store.tree(depth=1)` method, fetching all existing metadata docs at the root level and one level below.